database/data_seeding/nosql_seeder.js

```diff
@@ -2,45 +2,44 @@ const { user, password } = require('../../database_configs/nosql_database.config
 const nano = require('nano')(`http://${user}:${password}@localhost:5984`);
 const { generateData } = require('./data_generator.js');
 
-const destroyDb = async (dbName) => {
-  await nano.db.destroy(dbName);
-}
+const seedNoSqlData = async (numRecords, pace) => {
+
+  if ((numRecords / pace) - Math.floor(numRecords / pace) !== 0) {
+    throw new Error('Number of records must be divisible by pace (ex: 1,000 records at a pace of 100)');
+  }
+
+  const numBatches = numRecords / pace;
+  const copyOfNumRecords = numRecords;
+
+  try {
+    await nano.db.destroy('product_features');
+    await nano.db.create('product_features');
+  } catch (err) {
+    throw new Error('Error initializing database: ', err);
+  }
+
+  const pf = nano.use('product_features');
+
+  let currentBatch = 1;
+  let startingId = 1;
 
-const createDb = async (dbName) => {
-  await nano.db.create(dbName)
+  while (numRecords > 0) {
+    const data = generateData(1000, startingId, 'nosql');
+
+    await pf.bulk({ docs: data })
     .then((res) => {
-      console.log('Database successfully created', res);
+      console.log(`Batch number ${currentBatch} / ${numBatches} successfully seeded for IDs ${startingId} through ${(startingId + pace) - 1}`);
     })
     .catch((err) => {
-      console.log('Error creating database', err);
+      throw new Error(`Error seeding database on batch number ${currentBatch} / ${numBatches} for IDs ${startingId} through ${(startingId + pace) - 1}`);
     });
-}
-
-const insertData = (data) => {
-  // code
-}
 
-const seedNoSqlData = async (numRecords, pace) => {
-
-  await destroyDb('product_features');
-  await createDb('product_features');
-  const pf = nano.use('product_features');
+    numRecords -= pace;
+    startingId += pace;
+    currentBatch++;
+  }
 
-  // while (numRecords < 0) {
-  //   // seed
-  // }
-  const data = generateData(5, 1, 'nosql');
-
-  data.map((record) => {
-    const { productIdEncid } = record;
-    pf.insert(record, productIdEncid)
-      .then((res) => {
-        console.log('Inserted record successfully', res);
-      })
-      .catch((err) => {
-        console.log('Error inserting record', err);
-      });
-  });
+  console.log(`Database seeding for ${copyOfNumRecords} records complete`);
 }
 
 seedNoSqlData(10000000, 1000);
```

Comment:
is 1,000 the largest batch insert that can be done using "bulk" with nano or is it just a nice round number that you picked?

Reply:
1,000 is the array size for bulk data to insert that I originally chose based on what Emmanuel was saying regarding not having too much in memory to store the array, I later bumped it up to 2,000 to test if seeding would go faster but the insert time just doubled so 1,000 took just as long as 2,000, it really comes down to how much data  you wanna/can store in memory every time you're generating that batch to insert